pkg/kubelet/kubelet.go

```diff
@@ -1076,6 +1070,13 @@ func (kl *Kubelet) SyncPods(pods []api.BoundPod) error {
 			// syncPod() will handle this one.
 			continue
 		}
+		_, _, podAnnotations := ParsePodFullName(podFullName)
+		if source := podAnnotations[ConfigSourceAnnotationKey]; !kl.sourceReady(source) {
```

Comment:
Yep, if ```ParsePodFullName(...)``` is unable to parse the container name then ```podAnnotations``` will be ```map[string]string(nil)``` and thus ```source``` will be empty string. And ```sourcesSeen.HasAll``` will return false indicating it has never seen the empty string source before, thus the container will not be killed.